web/src/buddy_data.js

```diff
@@ -24,8 +30,35 @@ import * as util from "./util";
 export const max_size_before_shrinking = 600;
 
 let is_searching_users = false;
+
+// TODO: This should ideally be in buddy_list.js since it uses jquery,
+// but can't be easily placed there due to import cycles.
+function update_empty_list_placeholders() {
+    const empty_list_message = is_searching_users
+        ? $t({defaultMessage: "No matching users."})
+        : $t({defaultMessage: "None."});
+
+    $("#buddy-list-other-users").data("search-results-empty", empty_list_message);
+    if ($("#buddy-list-other-users .empty-list-message").length) {
```

Comment:
OK. I don't love using `.data("search-results-empty", empty_list_message)` as a setter; prefer `.attr("data-search-results-empty", ...)`. The `.data` function is evil in that it reads a `data-search-results-empty` field, but writes a thing that is only read by `jQuery`, so any use of this is both harder to grep (as you might have a `data-search-results-empty` in the DOM and also makes migrating off `jQuery` harder.

Posted about https://chat.zulip.org/#narrow/stream/6-frontend/topic/jQuery.20.2Edata/near/1686087 since we should just discuss and potentially lint-ban this as a project.

Reply:
When I change it, it doesn't work anymore. I'll follow up on CZO.